classes/core/PKPRoutingProvider.php

```diff
@@ -0,0 +1,181 @@
+        DecodeApiTokenWithValidation::class,
+        ValidateCsrfToken::class,
+        RemoveTrailingSlash::class,
+        ValidatePostSize::class,
+        TrimStrings::class,
+        ConvertEmptyStringsToNull::class,
+    ];
+
+    /**
+     * The application's route middleware.
+     * These middleware can/should be assigned to specific routes or routes groups individually.
+     */
+    protected array $routeMiddlewares = [
+        'has.roles'     => HasRoles::class,
+        'has.user'      => HasUser::class,
+        'has.context'   => HasContext::class,
+    ];
+
+    public static function getGlobalRouteMiddlewares(): array
+    {
+        return self::$globalMiddlewares;
+    }
+
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        $this->registerRouter();
+        $this->registerRouteMiddlewares();
+        $this->registerRoutePatterns();
+        $this->registerUrlGenerator();
+        $this->registerRedirector();
+        $this->registerResponseBindings();
+        $this->registerPsrRequest();
+        $this->registerPsrResponse();
+        $this->registerResponseFactory();
+        $this->registerCallableDispatcher();
+        $this->registerControllerDispatcher();
+    }
+
+    public function registerRouter(): void
+    {
+        $this->app->singleton('router', function ($app) {
+            $request = Request::capture();
+            $app->instance(\Illuminate\Http\Request::class, $request);
+            return new Router($app['events'], $app);
+        });
+    }
+
+    public function registerRouteMiddlewares(): void
+    {
+        $router = app('router'); /** @var \Illuminate\Routing\Router $router */
+
+        foreach ($this->routeMiddlewares as $key => $middleware) {
+            $router->aliasMiddleware($key, $middleware);
+        }
+    }
+
+    public function registerRoutePatterns(): void
+    {
+        $router = app('router'); /** @var \Illuminate\Routing\Router $router */
+
+        $router->pattern('contextPath', '(.*?)');
+        $router->pattern('version', '(.*?)');
+    }
+
+    protected function registerResponseBindings(): void
+    {
+        $container = PKPContainer::getInstance();
+
+        $container->bind(\Illuminate\Routing\RouteCollectionInterface::class, \Illuminate\Routing\RouteCollection::class);
+        $container->bind(
+            \Illuminate\View\ViewFinderInterface::class, 
+            fn ($app) => new \Illuminate\View\FileViewFinder(app(\Illuminate\Filesystem\Filesystem::class), [])
+        );
+        $container->bind(\Illuminate\Contracts\View\Factory::class, \Illuminate\View\Factory::class);
+        $container->bind(\Illuminate\Contracts\Routing\ResponseFactory::class, \Illuminate\Routing\ResponseFactory::class);
+    }
+
+    /**
+     * Register the URL generator service.
+     *
+     * @return void
+     */
+    protected function registerUrlGenerator()
```

Comment:
Is this a copy from the parent class?

Reply:
The only reason for this is because laravel `app('request')` expects to return an instance `Illuminate\Http\Request` but because of our [override](https://github.com/pkp/pkp-lib/blob/main/classes/core/PKPContainer.php#L163), now the `app('request')` return an instance of our `PKPRequest` so I had to override it with `Request::capture()` . 

I am wondering if the [override](https://github.com/pkp/pkp-lib/blob/main/classes/core/PKPContainer.php#L163) is necessary at all ? I don't see any usage of of `app('request')` or `app()->get('request')` in our codebase but not too sure about the plugins . If not, we can clean it up a lot .

Reply:
OK, great!